### ISSUE TITLE
remnant pip after uninstalling python 

### ISSUE THREAD

**Issue by HumairaFasih** [Homebrew/homebrew-core#132135]
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
brew config      
humairafasih@Humairas-MacBook-Pro
HOMEBREW_VERSION: 4.0.19
ORIGIN: https://github.com/Homebrew/brew
HEAD: 4446fe603cc19316036fd07e5df1124c4d15ead2
Last commit: 5 days ago
Core tap JSON: 27 May 06:46 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 14.0.3 build 1403
Git: 2.39.2 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.84.0 => /usr/bin/curl
macOS: 13.0.1-arm64
CLT: 14.3.0.0.1.1679647830
Xcode: N/A
Rosetta 2: false

brew doctor
Your system is ready to brew.
```

### Verification

- [X] My "`brew doctor` output" says `Your system is ready to brew.` and am still able to reproduce my issue.
- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.
- [X] I searched for recent similar issues at https://github.com/Homebrew/homebrew-core/issues?q=is%3Aissue and found no duplicates.

### What were you trying to do (and why)?

I was trying to set up different python versions and understand how pip worked with those versions in homebrew, since understanding this helps debug versioning issues later.

### What happened (include all command output)?

I installed python 3.10 using `brew install python@3.10`, which completed successfully. This also installed pip3.10 into /opt/homebrew/bin
I then proceeded to uninstall python3.10 using `brew uninstall python3.10` which also completed successfully. See below for proof that I do not have python3.10 in my system now.

<img width="388" alt="Screenshot 2023-05-27 at 12 48 40 PM" src="https://github.com/Homebrew/homebrew-core/assets/66149055/4413d679-e7b9-40f4-8414-2350639c12a1">

However, pip3.10 was not removed from the system automatically. 
Attached below is the output of commands where I checked the location of pip3.10 (which confirms it is still present), checked its version (which resulted in a bad interpreter error since python3.10 has been uninstalled), and then attempted to uninstall pip3.10, which didn't work and reported that no such formula is available.

<img width="569" alt="terminal output for pip bug" src="https://github.com/Homebrew/homebrew-core/assets/66149055/2c677ba2-9f87-4f41-8ba7-2124124a861a">

### What did you expect to happen?

It would be expected that any packages/binaries (pip3.10 in this case) that accompany the installation of another package (python@3.10) would automatically be cleaned up when the main installation is uninstalled.

Furthermore, if pip3.10 wasn't properly removed, the `brew uninstall pip3.10` command should work.
In my case, forceful removal of an executable via e.g. `rm -f` would work, but this should not be required. Optimal way would be for homebrew to support automatic uninstallation of the relevant pip version when its python version is uninstalled by the user.

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
Note: Please ensure you dont already have any pre-installs of python3.10 and pip3.10 
(check using `which python3.10` and `which pip3.10`)

Then reproduce using:
`brew install python@3.10`
`brew uninstall python@3.10`
`which pip3.10`
`pip3.10 --version`
`brew uninstall pip3.10`
```

**Comment by gromgit** [comment]
Unable to replicate:
```
% brew info python@3.10 
==> python@3.10: stable 3.10.11 (bottled)
Interpreted, interactive, object-oriented programming language
https://www.python.org/
Not installed
[...]

% brew install !$
brew install python@3.10
==> Downloading https://formulae.brew.sh/api/formula.jws.json
########################################################################################################################################################################################### 100.0%
==> Downloading https://formulae.brew.sh/api/cask.jws.json
########################################################################################################################################################################################### 100.0%
==> Fetching python@3.10
[...]
==> Summary
🍺  /opt/homebrew/Cellar/python@3.10/3.10.11: 3,099 files, 57.2MB
==> Running `brew cleanup python@3.10`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).

% which -a pip3.10
/opt/homebrew/bin/pip3.10

% brew uninstall python@3.10
Uninstalling /opt/homebrew/Cellar/python@3.10/3.10.11... (3,099 files, 57.2MB)

% which -a pip3.10          
pip3.10 not found
```
What's the output of:
```
readlink /opt/homebrew/bin/pip3.10
ls -l /opt/homebrew/bin/pip3.10
```
?

**Comment by HumairaFasih** [comment]
<img width="571" alt="Screenshot 2023-05-27 at 1 40 05 PM" src="https://github.com/Homebrew/homebrew-core/assets/66149055/4a70e6eb-3ce9-4771-af83-9d2e278cf7f8">

The output you requested is the above^

Also, I tried reproducing with another python version e.g. 3.8 after your comment and it doesn't give an issue. Unsure of whether the red color of the binary has anything to do with broken symlinks or the like? If so, what does that signal? Because I have that color for `brew` and `pip` binaries in opt/homebrew/bin as well

**Comment by gromgit** [comment]
Please send text output _as text_ in future, instead of screenshots.

Your `/opt/homebrew/bin/pip3.10` is a _file_, not a symlink. That's why `brew uninstall` didn't remove it, because it's expecting `pip3.10` to be a symlink to `../Cellar/python@3.10/.../bin/pip3.10`.

`brew` is the only file that should be in `/opt/homebrew/bin`, everything else should be symlinks into `../Cellar/<something_or_other>`. Something other than Homebrew put `pip` and `pip3.10` files in there.

**Comment by gromgit** [comment]
Anyway, try `brew install python@3.10`, then `brew uninstall python@3.10`. I think that should properly get rid of the mysterious files.

**Comment by HumairaFasih** [comment]
I installed miniconda via bash script after this. But that puts its required binaries in a different location, so I'm not sure how `pip3.10` and `pip` and 'pyvenv` exist in opt/homebrew/bin as files.

<img width="498" alt="Screenshot 2023-05-27 at 7 19 42 PM" src="https://github.com/Homebrew/homebrew-core/assets/66149055/662963c6-bc89-4905-b0bd-6108853f3def">

And yes your solution worked! It probably created the symlink again, and then removed it when I did the reinstall. 
My `pip` is also like that tho, and so is pyvenv. Do I run `brew uninstall pip` and `brew uninstall pyvenv` ?
Will I run into issues if I let them exist as files instead of symlinks in this location?
